### PR TITLE
don't copy SSH key to master

### LINKED ISSUE #1209: don't copy SSH key
Currently, to SSH into EMR slave nodes, we first copy our SSH key to the master node.

It would be cleaner and more secure to use SSH agent forwarding. This might require a dummy config file, which we already make.

### PR DESCRIPTION
**Pull request by davidmarin**
When we needed to SSH into non-master nodes on EMR, we used to copy our key to the master node (which is potentially a security issue). This changes it so we add the key to the SSH agent instead. Fixes #1209.

### CODE REVIEW

**Comment by 88manpreet** [comment]
+1

File: tests/mock_boto3/case.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import itertools
 import os
 import shutil
+import stat
```
Comment (88manpreet) [inline tests/mock_boto3/case.py]: Missed this earlier. Where are we using this?

File: mrjob/fs/ssh.py [hunk 2]
```diff
@@ -88,12 +90,17 @@ def _ssh_cmd_args(self, address, cmd_args):
                 key_pair_file = self._remote_key_pair_file
```
Comment (88manpreet) [inline mrjob/fs/ssh.py]: I think we are never using this, right? (when i !=0)

File: tests/mock_boto3/case.py (hunk 1 above)
Comment (davidmarin) [reply tests/mock_boto3/case.py]: whoops, we're not. you can delete it (or remove it in pre-release pyflaking)

File: mrjob/fs/ssh.py (hunk 2 above)
Comment (davidmarin) [reply mrjob/fs/ssh.py]: Oh, right, `self._remote_key_pair_file` is unused and should be removed.

**Pull request merged**